### PR TITLE
poc work

### PR DESCRIPTION
**Pull request by odra**
To compile and run it, just run `make install`

### CODE REVIEW

File: pkg/apis/aerogear/v1alpha1/mobileclient_types.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	MobileClientType = metav1.TypeMeta{
+		APIVersion: "mobile.k8s.io/v1alpha1",
```
Comment (Boomatang) [inline pkg/apis/aerogear/v1alpha1/mobileclient_types.go]: The APIVersion has been changed in the lasted version of MDC. This should be `mdc.aerogear.org/v1alpha1` now.

File: res/mobileclient.yaml.tmpl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: mobile.k8s.io/v1alpha1
```
Comment (Boomatang) [inline res/mobileclient.yaml.tmpl]: The APIVersion has been changed in the lasted version of MDC. This should be `mdc.aerogear.org/v1alpha1` now.

File: pkg/ag/app/_testdata/mobileclient.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: mobile.k8s.io/v1alpha1
```
Comment (Boomatang) [inline pkg/ag/app/_testdata/mobileclient.yaml]: The APIVersion has been changed in the lasted version of MDC. This should be `mdc.aerogear.org/v1alpha1` now.

File: pkg/apis/aerogear/v1alpha1/register.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/scheme"
+)
+
+var (
+	// SchemeGroupVersion is group version used to register these objects
+	SchemeGroupVersion = schema.GroupVersion{Group: "mobile.k8s.io", Version: "v1alpha1"}
```
Comment (Boomatang) [inline pkg/apis/aerogear/v1alpha1/register.go]: The `GroupVersion` should been changed to work with the lasted version of MDC. This should be `mdc.aerogear.org` not `mobile.k8s.io`.

File: cmd/link/link_auth/helpers.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+			APIVersion: "servicecatalog.k8s.io/v1beta1",
+			Kind:       "ServiceBinding",
+		},
+		ObjectMeta: v12.ObjectMeta{
+			Annotations: map[string]string{
+				"binding.aerogear.org/consumer": appName,
+				"binding.aerogear.org/provider": instance.Name,
+			},
+			Labels: map[string]string{
+				"mobileclients.mobile.k8s.io/clientId": appName,
```
Comment (Boomatang) [inline cmd/link/link_auth/helpers.go]: his should be `mdc.aerogear.org` not `mobile.k8s.io`.

File: deploy/crds/mobile-client-crd.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: mobileclients.mobile.k8s.io
```
Comment (Boomatang) [inline deploy/crds/mobile-client-crd.yaml]: This should be `mdc.aerogear.org` not `mobile.k8s.io`.

**Review by Boomatang — CHANGES_REQUESTED** [review]
When running the `make test` I get an error in the e2e tests but at this early stage I feel this would be ok.

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+.PHONY: test/e2e
+test/e2e:
+	@kubectl apply -f deploy/crds
+	@go test -v -race -cover ./test/e2e/...
+
+.PHONY: test
+test: test/unit test/e2e
+
+.PHONY: build/cli
+build/cli:
```
Comment (craicoverflow) [inline Makefile:37]: Should this also consider the user's enviroment and build for that? For example, if I run `build/cli` on a Mac, will it work?

Comment (odra) [reply Makefile:37]: bin here is the local bin folder, other paths are using GOPATH and HOME env vars which should be ok (I don't have a mac so I need a mac user to test it)

**Comment by odra** [comment]
@Boomatang e2e tests requires you to be logged in as admin in a kubernetes instance, minikube is usually enough to run those tests

**Review by Boomatang — APPROVED** [review]
I think this is now in a good state that we could merge it into master. When testing locally all the unit & e2e tests passed. The manually test path I tried worked, there is a few little things. But I think we should merge it and create jiras around some of the issues.

**Pull request merged**